test/data_access/JSONPersistenceTest.java

```diff
@@ -0,0 +1,121 @@
+            folder.newFile(filePath);
+        } catch (IOException e) {
+            System.out.println("Couldn't make testing file");
+            fail();
+        }
+        /* Make the instance. */
+        RecipeAPI recipeDAO = new TheMealDB(recipeFactory);
+        this.instance = new JSONPersistence(userFactory, filePath, recipeDAO);
+    }
+
+    @Test
+    public void testEmptyUserOutput() throws IOException {
+        instance.save(emptyUser);
+        String fileContents = new String(Files.readAllBytes(Path.of(this.filePath)));
+        assertTrue(fileContents.contains("favourites"));
+        assertTrue(fileContents.contains("tags"));
+        assertTrue(fileContents.contains("{}"));
+        assertTrue(fileContents.contains("[]"));
+    }
+    @Test
+    public void testSingleUserOutput() throws IOException {
+        instance.save(filledUser);
+        String fileContents = new String(Files.readAllBytes(Path.of(this.filePath)));
+        /* Check that we have the right things in the contents of the files */
+        assertTrue(fileContents.contains("asdf"));
+        assertTrue(fileContents.contains("favourites"));
+        assertTrue(fileContents.contains("tags"));
+        assertTrue(fileContents.contains("tag"));
+        assertTrue(fileContents.contains("52802"));
+        assertTrue(fileContents.contains("52872"));
+        assertTrue(fileContents.contains("52855"));
+        assertTrue(fileContents.contains("52870"));
+    }
+
+    @Test
+    public void testReadingContents() throws IOException {
+        /* Write some testing data to the file */
+        PrintWriter file = new PrintWriter("temp.json");
+        file.println("{\n" +
+                "    \"favourites\": [\n" +
+                "        52802,\n" +
+                "        52872\n" +
+                "    ],\n" +
+                "    \"username\": \"asdf\",\n" +
+                "    \"tags\": {\"tag\": [\n" +
+                "        52855,\n" +
+                "        52870\n" +
+                "    ]}\n" +
+                "}");
+        Set<Integer> actualFavouriteIds = Set.of(52802, 52872);
+        Set<Integer> actualTaggedIds = Set.of(52855, 52870);
+
+        User user = instance.load();
+        assertEquals(user.getUsername(), "asdf");
+        assertEquals(user.getFavourites().size(), 2);
+
+        Integer[] favouriteIds = new Integer[2];
+        for (int i = 0; i < user.getFavourites().size(); i++) {
+            favouriteIds[i] = user.getFavourites().get(i).getMealId();
+        }
+
+        Integer[] taggedIds = new Integer[2];
+        ArrayList<Recipe> taggedRecipes = user.getTaggedRecipes().get("tag");
+        for (int i = 0; i < taggedRecipes.size(); i++) {
+            taggedIds[i] = taggedRecipes.get(i).getMealId();
+        }
+
+        assertEquals(favouriteIds.length, 2);
+        assertTrue(actualFavouriteIds.containsAll(List.of(favouriteIds)));
+        assertTrue(actualTaggedIds.containsAll(List.of(taggedIds)));
+    }
+}
```

Comment:
Another test to add here would be a test that ensures the JSONException is thrown if we try to read an empty JSON file or a JSON file that is not formatted in the way in which we expect.